rt-pwa-extensions.php

```diff
@@ -69,3 +69,60 @@ function( $headers ) {
 		return $headers;
 	}
 );
+
+
+// Register different image sizes for icons.
+$icon_sizes = [ '72', '96', '128', '144', '152', '192', '384', '512' ];
+
+foreach ( $icon_sizes as $size ) {
+	// Generete a image size label: pwa-icon-<size>.
+	$image_size_label = sprintf( 'pwa-icon-%1$d', $size );
+	add_image_size( $image_size_label, absint( $size ), absint( $size ) );
```

Comment:
I think this array of size will be added for all the image uploads. 
Meaning every image upload will create these images. 
We need to find some better solution to only generate these sizes for `PWA icon` not for other images.

@dharmin 

@chandrapatel Any thoughts?

Reply:
Can we use `wp_get_attachment_image_src ` without using add_image_size for 

> $icon_sizes = [ '72', '96', '128', '144', '152', '192', '384', '512' ];

Maybe by passing custom array sizes like this instead of `add_image_size`.

Reply:
>Can we use wp_get_attachment_image_src without using add_image_size for

Nope. If we pass custom image sizes but if image not generated for those images then it will return full image.

May be we can use [wp_get_image_editor()](https://developer.wordpress.org/reference/functions/wp_get_image_editor/) function to create images with different sizes whenever site icon change?